lib/rspec/support/method_signature_verifier.rb

```diff
@@ -282,7 +305,7 @@ class MethodSignatureVerifier
 
       def initialize(signature, args=[])
         @signature = signature
-        @non_kw_args, @kw_args = split_args(*args)
+        @non_kw_args, @kw_args = split_args(args.clone)
         @min_non_kw_args = @max_non_kw_args = @non_kw_args
         @arbitrary_kw_args = @unlimited_args = false
       end
```

Comment:
something had to change here.  Splatting the args meant the special marking for the kwargs hash gets lost.  This doesnt happen to functions with the ruby2_keywords... except it won't apply here because this function doesn't take the *args input.  So either the args array is cloned (which persists the special flagging) or the input to this function needs to swap to using *args.  I figured cloning was the less obtrusive change.